modules/library/main/src/main/java/org/geotools/filter/expression/PropertyAccessors.java

```diff
@@ -44,18 +45,15 @@
         // reasons         
         cache.add( new NullPropertyAccessorFactory()); //NC - added       
         cache.add( new SimpleFeaturePropertyAccessorFactory());
-        cache.add( new DirectPropertyAccessorFactory());       
-        Iterator factories = FactoryRegistry
-                 .lookupProviders(PropertyAccessorFactory.class);
-         while (factories.hasNext()) {
-            Object factory = factories.next();
-            if ( factory instanceof SimpleFeaturePropertyAccessorFactory || factory instanceof DirectPropertyAccessorFactory
-                 || factory instanceof NullPropertyAccessorFactory )
-                continue;
-            
-            cache.add((PropertyAccessorFactory) factory);
-         }
-         FACTORY_CACHE = (PropertyAccessorFactory[]) cache.toArray(new PropertyAccessorFactory[cache.size()]);
+        cache.add( new DirectPropertyAccessorFactory());
+        for (PropertyAccessorFactory factory : ServiceLoader.load(PropertyAccessorFactory.class)) {
+            boolean nonCached = factory instanceof SimpleFeaturePropertyAccessorFactory
+                    || factory instanceof DirectPropertyAccessorFactory
+                    || factory instanceof NullPropertyAccessorFactory;
+            if (!nonCached)
+                cache.add(factory);
```

Comment:
Thanks that is a clear way to describe the cache condition. We tend to use braces even for single line if statements

Reply:
Done.